Core/CoreDependencyProvider.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  CoreDependencyProvider.swift
+//  DuckDuckGo
+//
+//  Copyright © 2022 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public protocol CoreDependencyProvided {
+    var bookmarksCachingSearch: BookmarksCachingSearch { get }
+}
+
+/// Provides dependencies for objects that are not directly instantiated
+/// through `init` call
+public class CoreDependencyProvider: CoreDependencyProvided {
```

Comment:
This is interesting, how do you plan on using it in future?

Reply:
I saw AppDependencyProvider already in place and thought Core needed something similar in this case; ideally we would move our singletons such as Favicons to this DependencyProvider for DI in the future